src/modules/storage/reference.js

```diff
@@ -96,7 +97,7 @@ export default class StorageReference extends ReferenceBase {
    * @return {Promise}
    */
   putFile(filePath: Object, metadata: Object = {}): StorageTask {
-    let _filePath = filePath.replace('file://', '');
+    let _filePath = isIOS ? filePath.replace('file://', '') : filePath;
     if (_filePath.includes('%')) _filePath = decodeURI(_filePath);
```

Comment:
Can you change `decodeURI ` to `decodeURIComponent` - this will fix a bug I noticed on v6